test/org/javarosa/xform/parse/XFormParserTest.java

```diff
@@ -108,13 +109,18 @@ public void parsesPreloadForm() throws IOException {
         assertEquals("Sample Form - Preloading", formDef.getTitle());
     }
 
-    @Test public void parsesSecondaryInstanceForm() throws IOException, XPathSyntaxException {
+    @Test public void parsesSecondaryInstanceForm() throws IOException {
         FormDef formDef = parse(SECONDARY_INSTANCE_XML).formDef;
         assertEquals("Form with secondary instance", formDef.getTitle());
     }
 
+    private String allButLastSubpath(Path formName) {
+        return formName.subpath(0, formName.getNameCount() - 1).toString();
```

Comment:
This can be `formName.getParent().toString()`, and I think it could be inlined (no need for a method for that, I think)

Reply:
I missed that when I looked up `subpath`. Thanks!